pkg/dbsyncers/placement_syncer.go

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2021 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package dbsyncers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"k8s.io/apimachinery/pkg/api/errors"
+	clustersv1beta1 "open-cluster-management.io/api/cluster/v1beta1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	placementsSpecTableName   = "placements"
+	placementsStatusTableName = "placements"
+)
+
+func addPlacementStatusDBSyncer(mgr ctrl.Manager, databaseConnectionPool *pgxpool.Pool,
+	syncInterval time.Duration) error {
+	err := mgr.Add(&genericDBSyncer{
+		syncInterval: syncInterval,
+		syncFunc: func(ctx context.Context) {
+			syncPlacements(ctx,
+				ctrl.Log.WithName("placement-db-syncer"),
+				databaseConnectionPool,
+				mgr.GetClient())
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to add placements syncer to the manager: %w", err)
+	}
+
+	return nil
+}
+
+func syncPlacements(ctx context.Context, log logr.Logger, databaseConnectionPool *pgxpool.Pool,
+	k8sClient client.Client) {
+	log.Info("performing sync of placement-status")
+
+	rows, err := databaseConnectionPool.Query(ctx,
+		fmt.Sprintf(`SELECT payload->'metadata'->>'name', payload->'metadata'->>'namespace' 
+		FROM spec.%s WHERE deleted = FALSE`, placementsSpecTableName))
+	if err != nil {
+		log.Error(err, "error in getting placement spec")
+		return
+	}
+
+	for rows.Next() {
+		var name, namespace string
+
+		err := rows.Scan(&name, &namespace)
+		if err != nil {
+			log.Error(err, "error in select", "table", placementsSpecTableName)
+			continue
+		}
+
+		go handlePlacementStatus(ctx, log, databaseConnectionPool, k8sClient, name, namespace)
+	}
+}
+
+func handlePlacementStatus(ctx context.Context, log logr.Logger, databaseConnectionPool *pgxpool.Pool,
+	k8sClient client.Client, placementName string, placementNamespace string) {
+	log.Info("handling a placement", "name", placementName, "namespace", placementNamespace)
+
+	placement, err := getAggregatedPlacements(ctx, databaseConnectionPool, placementName, placementNamespace)
+	if err != nil {
+		log.Error(err, "failed to get aggregated placement", "name", placementName, "namespace", placementNamespace)
+		return
+	}
+
+	if placement == nil {
```

Comment:
I didn't get this if.
one possible scenario is where nil is returned when there is an error.
in that case the err != nil handles it.

the other scenario is when there are no matching rows. in this scenario we should clean status.
we shouldn't return nil in that scenario.